CRM/Civicase/Hook/SelectWhereClause/RestrictSearchToAccessibleCaseCategories.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * Restrict search to accessible case categories if case entity is involved.
+ */
+class CRM_Civicase_Hook_SelectWhereClause_RestrictSearchToAccessibleCaseCategories {
```

Comment:
This name `RestrictSearchToAccessibleCaseCategories` is not appropriate again because this hook will do more than restrict search. It will affect more than search page and will affect selecting cases via api, ui, or searches. Everywhere the Case entity is being called will be affected. Maybe a name like `LimitCaseQueryToAccessibleCaseCategories` or something better.